edb/server/protocol/binary.pyx

```diff
@@ -526,7 +547,21 @@ cdef class EdgeConnection(frontend.FrontendConnection):
             self.debug_print('Extra variables', source.variables(),
                              'after', source.first_extra())
 
-        return await dbv.parse(query_req)
+        query_unit_group = dbv.lookup_compiled_query(query_req)
+
+        # If we have to do a compile within a transaction, suppress
+        # the idle_in_transaction_session_timeout.
+        suppress_timeout = (
+            dbv.in_tx() and not dbv.in_tx_error() and query_unit_group is None
+        )
+        if suppress_timeout:
+            await self._suppress_tx_timeout()
```

Comment:
What do we think about doing a `loop.call_at` half the current timeout here instead?  Too fiddly?

Reply:
I think we shouldn't do that because:
> The problem with the possibly delayed approach is that if a bunch of the timeout has already elapsed when the query is sent, then we can still timeout while compiling.